core/src/main/scala/com/salesforce/op/filters/RawFeatureFilterResults.scala

```diff
@@ -0,0 +1,140 @@
+
+  /**
+   * RawFeatureFilterResults to json
+   *
+   * @param results raw feature filter results
+   * @return json array
+   */
+  def toJson(results: RawFeatureFilterResults): String = Serialization.write[RawFeatureFilterResults](results)
+
+  /**
+   * RawFeatureFilterResults from json
+   *
+   * @param json json
+   * @return raw feature filter results
+   */
+  def fromJson(json: String): Try[RawFeatureFilterResults] = Try {
+    Serialization.read[RawFeatureFilterResults](json)
+  }
+
+}
+
+/**
+ * Contains configuration settings for Raw Feature Filter
+ */
+case class RawFeatureFilterConfig
+(
+  minFill: Double = 0.0,
+  maxFillDifference: Double = Double.PositiveInfinity,
+  maxFillRatioDiff: Double = Double.PositiveInfinity,
+  maxJSDivergence: Double = 1.0,
+  maxCorrelation: Double = 1.0,
+  correlationType: CorrelationType = CorrelationType.Pearson,
+  jsDivergenceProtectedFeatures: Set[String] = Set.empty,
+  protectedFeatures: Set[String] = Set.empty
+) extends RawFeatureFilterConfigLike
+
+/**
+ * Contains raw feature metrics computing in Raw Feature Filter
+ *
+ * @param name                          feature name
+ * @param trainingFillRate              proportion of values that are null in the training distribution
+ * @param trainingNullLabelAbsoluteCorr correlation between null indicator and the label in the training distribution
+ * @param scoringFillRate               proportion of values that are null in the scoring distribution
+ * @param jsDivergence                  Jensen-Shannon (JS) divergence between the training and scoring distributions
+ * @param fillRateDiff                  absolute difference in fill rates between the training and scoring distributions
+ * @param fillRatioDiff                 ratio of difference in fill rates between the training and scoring distributions
+ */
+case class RawFeatureFilterMetrics
+(
+  name: String,
+  trainingFillRate: Double,
+  trainingNullLabelAbsoluteCorr: Option[Double],
+  scoringFillRate: Option[Double],
+  jsDivergence: Option[Double],
+  fillRateDiff: Option[Double],
+  fillRatioDiff: Option[Double]
+) extends RawFeatureFilterMetricsLike
+
+/**
+ * Contains results of Raw Feature Filter tests for a given feature
+ *
+ * @param name                    feature name
+ * @param trainingUnfilledState   training fill rate did not meet min required
+ * @param trainingNullLabelLeaker null indicator correlation (absolute) exceeded max allowed
+ * @param scoringUnfilledState    scoring fill rate did not meet min required
+ * @param jsDivergenceMismatch    distribution mismatch: JS Divergence exceeded max allowed
+ * @param fillRateDiffMismatch    distribution mismatch: fill rate difference exceeded max allowed
+ * @param fillRatioDiffMismatch   distribution mismatch: fill ratio difference exceeded max allowed
+ * @param excluded                feature excluded after failing one or more tests
+ */
+case class ExclusionReasons
```

Comment:
Unfortunately this is a tough situation because need to make `ExclusionReasons` names distinct from the those in `RawFeatureFilterMetrics` (where I've already used many of the names you're suggesting!):

```
case class RawFeatureFilterMetrics
(
  name: String,
  trainingFillRate: Double,
  trainingNullLabelAbsoluteCorr: Option[Double],
  scoringFillRate: Option[Double],
  jsDivergence: Option[Double],
  fillRateDiff: Option[Double],
  fillRatioDiff: Option[Double]
) extends RawFeatureFilterMetricsLike
```

Thought really long about this... part of rationale for name choices was that these variables refer to outcomes of a comparison test (boolean), so don't want to make it seem like they are referring to a continuous value

Reply:
Ok, I'm not sure I follow why the field names in `RawFeatureFilterMetrics` need to be different from the ones in `ExclusionReasons`. It should be clear from the type which case class one is referring to, no?

Your point about them being booleans for test results makes sense though, so lets keep them as is.